mousestyles/path_diversity/path_features_advanced.py

```diff
@@ -0,0 +1,287 @@
+    path_obj : pandas.DataFrame
+        CX and CY must be contained.
+        The length must be greater than 2.
+
+    Returns
+    -------
+    edge points : dictionary
+        contains edge points the path which are
+        'xmin', 'xmax', 'ymin', and 'ymax'
+
+    Examples
+    --------
+    >>> movement = data.load_movement(1,2,1)
+    >>> sep = path_index(movement, 1, 1)
+    >>> path = movement[sep[2][0]:sep[2][1]+1]
+    >>> edges = find_edge_points(path)
+    """
+    if not isinstance(path_obj, pd.core.frame.DataFrame):
+        raise TypeError("path_obj must be pandas DataFrame")
+
+    if not set(path_obj.keys()).issuperset(['x', 'y']):
+        raise ValueError("the keys of path_obj must contain 'x', 'y'")
+
+    if len(path_obj) <= 1:
+        raise ValueError("path_obj must contain at least 2 rows")
+
+    return( {'xmin' : np.min(path_obj.x) , 'xmax' : np.max(path_obj.x) , \
+             'ymin' : np.min(path_obj.y) , 'ymax' : np.max(path_obj.y)} )
+
+def compute_area_rectangle(edge_points):
+    r"""
+    Returns the area of the rectangle spanned by the path.
+
+    The rectangle, constructed by edge_points which is supposed
+    to have xmin, xmax, ymin, and ymax, has 4 edges which are
+    1) (xmin, ymin)
+    2) (xmin, ymax)
+    3) (xmax, ymin)
+    4) (xmax, ymax)
+
+    The path which has such edge points is by design completely
+    covered by such rectangle.
+
+    Parameters
+    ----------
+    edge_points : dictionary
+        key must be xmin, xmax, ymin, ymax.
+        Expecting the output of find_edge_points.
+
+    Returns
+    -------
+    area : numpy float object
+        area of the rectangle spanned by the path.
+
+    Examples
+    --------
+    >>> movement = data.load_movement(1,2,1)
+    >>> sep = path_index(movement, 1, 1)
+    >>> path = movement[sep[2][0]:sep[2][1]+1]
+    >>> edges = find_edge_points(path)
+    >>> area_rec = compute_area_rectangle(edges)
+    """
+    if not isinstance(edge_points, dict):
+        raise TypeError("edge_points must be a dictionary")
+
+    if set(edge_points.keys()) != {'xmax', 'xmin', 'ymax', 'ymin'}:
+        raise ValueError("the keys of edge_points must be 'xmax', 'xmin', 'ymax', and 'ymin'")
+
+    return( (edge_points['xmax']-edge_points['xmin']) * \
+           (edge_points['ymax']-edge_points['ymin']) )
+
+def find_center(edge_points):
+    r"""
+    Returns the center point of the path.
+
+    The center point is defined by the intersection point of
+    2 diagonal lines of the rectangle which is constructed by
+    edge_points. That is, for the elements of edge_points,
+    xmin, xmax, ymin, and ymax, the center point is expressed by
+    
+    x_center = (xmin + xmax)/2,
+    y_center = (ymin + ymax)/2.
+
+    Parameters
+    ----------
+    edge_points : dictionary
+        its key must be 'xmin', 'xmax', 'ymin', 'ymax'.
+        Expecting the output of find_edge_points.
+
+    Returns
+    -------
+    center point of the path : dictinoary
```

Comment:
just a simple typo (should be dictionary) but it might be important since it's a type